ui/themes/materialplus/password.go

```diff
@@ -0,0 +1,181 @@
+package materialplus
+
+import (
+	"image/color"
+
+	"gioui.org/layout"
+	"gioui.org/unit"
+	"gioui.org/widget"
+	"gioui.org/widget/material"
+
+	"github.com/raedahgroup/godcr-gio/ui"
+)
+
+type colors struct {
+	cancelLabelColor            color.RGBA
+	createButtonBackgroundColor color.RGBA
+}
+
+type editor struct {
+	widget   *widget.Editor
+	material material.Editor
+	line     *Line
+}
+
+// Password represents a form for collecting user password
+type Password struct {
+	colors     colors
+	titleLabel material.Label
+
+	spendingEditor *editor
+	confirmEditor  *editor
+
+	cancelButton         *widget.Button
+	createButton         *widget.Button
+	cancelButtonMaterial material.Button
+	createButtonMaterial material.Button
+
+	errorLabel material.Label
+}
+
+// Password initializes and returns an instance of Password
+func (t *Theme) Password() *Password {
+	cancelButtonMaterial := t.Button("Cancel")
+	cancelButtonMaterial.Background = ui.WhiteColor
+	cancelButtonMaterial.Color = ui.LightBlueColor
+
+	errorLabel := t.Body2("")
+	errorLabel.Color = ui.DangerColor
+
+	spendingEditor := &editor{
+		widget:   new(widget.Editor),
+		material: t.Editor("Spending password"),
+		line:     t.Line(),
+	}
+	confirmEditor := &editor{
+		widget:   new(widget.Editor),
+		material: t.Editor("Confirm spending password"),
+		line:     t.Line(),
+	}
+
+	confirmEditor.line.Color = ui.LightBlueColor
+	spendingEditor.line.Color = ui.LightBlueColor
+
+	p := &Password{
+		titleLabel: t.H5("Create spending password"),
+		errorLabel: errorLabel,
+
+		confirmEditor:        confirmEditor,
+		spendingEditor:       spendingEditor,
+		cancelButton:         new(widget.Button),
+		createButton:         new(widget.Button),
+		cancelButtonMaterial: cancelButtonMaterial,
+		createButtonMaterial: t.Button("Create"),
+
+		colors: colors{},
+	}
+
+	return p
+}
+
+func (p *Password) updateColors() {
+	p.colors.cancelLabelColor = ui.GrayColor
+	p.colors.createButtonBackgroundColor = ui.GrayColor
+
+	if p.bothPasswordsMatch() && p.confirmEditor.widget.Len() > 0 {
+		p.colors.createButtonBackgroundColor = ui.LightBlueColor
+	}
+}
+
+func (p *Password) processButtonClicks(gtx *layout.Context, createFunc func(string), cancelFunc func()) {
+	for p.createButton.Clicked(gtx) {
+		if p.validate() {
+			createFunc(p.spendingEditor.widget.Text())
+		}
+	}
+
+	for p.cancelButton.Clicked(gtx) {
+		p.Reset()
+		cancelFunc()
+	}
+}
+
+// Draw renders the widget to screen
```

Comment:
What do `createFunc` and `cancelFunc` do? (Seems obvious but add it for the sake of documentation)